zebra-chain/src/work/difficulty.rs

```diff
@@ -290,6 +290,28 @@ impl CompactDifficulty {
 
         Ok(difficulty)
     }
+
+    /// Returns a floating-point number representing a difficulty as a multiple
+    /// of the minimum difficulty for the provided network.
+    pub fn relative_to_network(&self, network: &Network) -> f64 {
+        let network_difficulty = network.target_difficulty_limit().to_compact();
+
+        let [mut n_shift, ..] = self.0.to_be_bytes();
+        let [n_shift_amount, ..] = network_difficulty.0.to_be_bytes();
+        let mut d_diff = f64::from(network_difficulty.0 << 8) / f64::from(self.0 << 8);
+
+        while n_shift < n_shift_amount {
+            d_diff *= 256.0;
+            n_shift += 1;
+        }
+
+        while n_shift > n_shift_amount {
+            d_diff /= 256.0;
+            n_shift -= 1;
+        }
+
+        d_diff
+    }
```

Comment:
This was copied from zcashd here: https://github.com/zcash/zcash/blob/master/src/rpc/blockchain.cpp#L34-L74.

Reply:
Should we credit it in a comment or something?

Reply:
Credited in a comment in https://github.com/ZcashFoundation/zebra/pull/8967/commits/37ddd6e19422c56cea6ed0f7742ba7b299179e75.

@nuttycom or @str4d, could one of you explain what this code is doing or point me to an explanation if one is already documented?

Reply:
Maybe leave a comment in the code to explain the code and resolve this?
https://github.com/ZcashFoundation/zebra/pull/8967/files#r1834294181